cli/cycpp.py

```diff
@@ -1305,8 +1352,34 @@ def impl(self, ind="  "):
             if 'derived_init' in info:
                 impl += ind + info['derived_init'] + '\n'
             else:
+                labels = info.get('alias', None)
+                if labels is None:
+                    labels = member if isinstance(t, STRING_TYPES) else [member]
                 reader = self.readers.get(t, self.readers.get(t[0], None))
-                impl += reader(member, alias, t, d, uitype, ind)
+
+                # generate condition to choose default vs given val if default exists
+                if d is not None:
+                    name = labels if isinstance(t, STRING_TYPES) else labels[0]
+                    impl += ind + 'if (sub->NMatches("{0}") > 0) {{\n'.format(name)
```

Comment:
Separately, *we* could clang-format code-gen'd code as a post-process (if
it exists on the system). Just a thought..

On Fri, Mar 13, 2015 at 9:59 AM, Robert Carlsen <notifications@github.com>
wrote:

> In cli/cycpp.py
> <https://github.com/cyclus/cyclus/pull/1098#discussion_r26392184>:
>
> >                  reader = self.readers.get(t, self.readers.get(t[0], None))
> > -                impl += reader(member, alias, t, d, uitype, ind)
> > +
> > +                # generate condition to choose default vs given val if default exists
> > +                if d is not None:
> > +                    name = labels if isinstance(t, STRING_TYPES) else labels[0]
> > +                    impl += ind + 'if (sub->NMatches("{0}") > 0) {{\n'.format(name)
>
> Fixed. Although I think it reasonable to assume people debugging codegen'd
> code just use something like astyle or clang-format. We shouldn't invest
> too much worrying about these solved problems.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1098/files#r26392184>.
>



-- 
Matthew Gidden
Ph.D. Candidate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192